utilities/xmlrpcpp/src/XmlRpcClient.cpp

```diff
@@ -295,19 +323,16 @@ XmlRpcClient::generateHeader(std::string const& body)
   header += _host;
 
   char buff[40];
-#ifdef _MSC_VER
-  sprintf_s(buff,40,":%d\r\n", _port);
-#else
-  sprintf(buff,":%d\r\n", _port);
-#endif
+  snprintf(buff,40,":%d\r\n", _port);
 
   header += buff;
   header += "Content-Type: text/xml\r\nContent-length: ";
 
+  // Windows and glibc have different modifier flags for printing size_t
```

Comment:
`%zu` should work across all platforms.

Reply:
No; `z` is a GNU-specific size specifier and is not listed in Microsoft's documentation: https://msdn.microsoft.com/en-us/library/tcxf1dw6.aspx

Reply:
I think the `z` is defined for C++11: http://en.cppreference.com/w/cpp/io/c/fprintf, and it appears to work on Windows (we use it all the time in ROS 2).

I tried this program out on http://webcompiler.cloudapp.net/:

```c++
#include <cstdio>

int main()
{
   size_t i = 42;
   printf("%zu\n", i);
   return 0;
}
```

The output was:

```
Compiled with  /EHsc /nologo /W4
main.cpp

Compilation successful!

Total compilation time: 328ms

42

Total execution time: 140ms
```

Reply:
We use C++11 internally but it was my understanding that this code still builds in C++03 mode to avoid forcing downstream packages to enable C++11, so I took extra steps to avoid using any C++11 features.

Reply:
The `.cpp` file can be compiled with C++11. We don't want to expose C++11 feature in the headers though since that requires all downstream packages to choose C++11 too.

Reply:
Since these changes are fixing stability issues with XmlRpcpp, it also my hope that they will be backported to the LTS versions of ROS.

Reply:
After they have been released into Lunar and had no regressions for a while the changes can be considered for backporting into Kinetic. But even for Kinetic a C++11 compiler is a requirement already. See [REP 3](http://www.ros.org/reps/rep-0003.html#kinetic-kame-may-2016-may-2021).

Reply:
That's a fair point, but given that all the modern MSVC compilers use C++11 by default (there is no `-std=c++11` like option), and all other compilers support `zu` without C++11 (that we care about), I think it's safe, in this instance, to just use `%zu`.

Reply:
Updated.

Reply:
`size_t` is *unsigned* so it needs to be `%zu` (not `%zd`).

Reply:
Updated.